pkg/apis/hco/v1beta1/hyperconverged_webhook.go

```diff
@@ -103,6 +104,33 @@ func (r *HyperConverged) ValidateCreate() error {
 
 func (r *HyperConverged) ValidateUpdate(old runtime.Object) error {
 	hcolog.Info("Validating update", "name", r.Name)
+
+	ctx := context.TODO()
+
+	oldR, ok := old.(*HyperConverged)
+	if !ok {
+		return fmt.Errorf("expect old object to be a %T instead of %T", oldR, old)
+	}
+
+	if !reflect.DeepEqual(
+		oldR.Spec.HyperConvergedConfig.InfraNodeSelector,
+		r.Spec.HyperConvergedConfig.InfraNodeSelector) ||
+		!reflect.DeepEqual(
+			oldR.Spec.HyperConvergedConfig.WorkloadNodeSelector,
+			r.Spec.HyperConvergedConfig.WorkloadNodeSelector) {
+		for _, obj := range []runtime.Object{
+			r.NewKubeVirt(),
+			r.NewCDI(),
+		} {
+			err := hcoutil.EnsureDeleted(cli, ctx, r.Name, obj, hcolog, true)
```

Comment:
Here we are trying to detect if we have existing workloads trying to delete in dryrun mode, but this is working only if we set BlockUninstallIfWorkloadsExist strategy.
Let's try to find a more robust mechanism.

Reply:
we can try a dry-run update on Kubevirt and CDI CRs that should validate it as well.